sdk/storage/Azure.Storage.DataMovement/src/Shared/DataMovementExtensions.cs

```diff
@@ -140,8 +140,8 @@ internal static StorageResourceItemProperties ToStorageResourceProperties(this F
             StreamToUriJobPart jobPart = StreamToUriJobPart.CreateJobPartFromCheckpoint(
                 job: baseJob,
                 partNumber: Convert.ToInt32(header.PartNumber),
-                sourceResource: sourceResource.GetStorageResourceReference(childSourceName),
-                destinationResource: destinationResource.GetStorageResourceReference(childDestinationName),
+                sourceResource: sourceResource.GetStorageResourceReference(childSourceName, default),
+                destinationResource: destinationResource.GetStorageResourceReference(childDestinationName, default),
```

Comment:
So, I think we need to do something other than `default` here? Since I _think_, as is, this would mean resuming a job would always result in Block Blobs for the source AND destination? Does the `JobPartPlanHeader` have source and destination resource ids in it? If so, maybe we use those here?

(This would apply to all job types in this file)

Reply:
Maybe we should also add some pause/resume tests around for Blob type.

Reply:
You're right. It looks like when the resume was done, we don't read the specific job part type (e.g. BlobDestinationCheckpointData) and that it's only done at the beginning. Looks like there was a false assumption that each the individual job parts would follow suit of the container checkpoint data, when that is not always the case.

There's a lot more plumbing here I have to do. So it might take me a while until the next update to this PR to be done. SInce some refactoring to how resume container transfer works needs to be done.

Reply:
Update made